dm/ctl/common/util.go

```diff
@@ -66,13 +66,83 @@ func PrintLines(format string, a ...interface{}) {
 
 // PrettyPrintResponse prints a PRC response prettily
 func PrettyPrintResponse(resp proto.Message) {
+	s, err := marshResponseToString(resp)
+	if err != nil {
+		PrintLines(errors.ErrorStack(err))
+	} else {
+		fmt.Println(s)
+	}
+}
+
+func marshResponseToString(resp proto.Message) (string, error) {
 	// encoding/json does not support proto Enum well
 	mar := jsonpb.Marshaler{EmitDefaults: true, Indent: "    "}
 	s, err := mar.MarshalToString(resp)
+	return s, errors.Trace(err)
+}
+
+// PrettyPrintResponseWithCheckTask prints a RPC response may contain response Msg with check-task's response prettily.
+// check-task's response may contain json-string when checking fail in `detail` field.
+// ugly code, but it is a little hard to refine this because needing to convert type.
+func PrettyPrintResponseWithCheckTask(resp proto.Message, subStr string) bool {
+	var (
+		err          error
+		found        bool
+		replacedStr  string
+		marshaledStr string
+		placeholder  = "PLACEHOLDER"
+	)
+	switch resp.(type) {
+	case *pb.StartTaskResponse:
+		sr := resp.(*pb.StartTaskResponse)
+		if strings.Contains(sr.Msg, subStr) {
+			found = true
+			rawMsg := sr.Msg
+			sr.Msg = placeholder // replace Msg with placeholder
+			marshaledStr, err = marshResponseToString(sr)
+			if err == nil {
+				replacedStr = strings.Replace(marshaledStr, placeholder, rawMsg, 1)
+			}
+		}
+	case *pb.UpdateTaskResponse:
+		sr := resp.(*pb.UpdateTaskResponse)
+		if strings.Contains(sr.Msg, subStr) {
+			found = true
```

Comment:
extract these logic code to a function

Reply:
a little hard to do that. these three proto.Message are `struct` but not `interface`, and we need to get/set their `Msg` field, so we need the exact type of them.

Reply:
is it ok?
```
switch sr := resp.(type) {
case xxx,yyy,zzz:
}
```

Reply:
not work. this can only **test** `type`, but we need **convert** type.

Reply:
updated to do the **convertion** in `switch` line, but cannot use multi cases in one Line (`case xxx,yyy,zzz:`).